sql/tables.go

```diff
@@ -393,20 +393,30 @@ type RewritableTable interface {
 	RewriteInserter(ctx *Context, oldSchema, newSchema PrimaryKeySchema, oldColumn, newColumn *Column, idxCols []IndexColumn) (RowInserter, error)
 }
 
-// AlterableTable should be implemented by tables that can receive ALTER TABLE statements to modify their schemas.
+// AlterableTable should be implemented by tables that can receive
+// ALTER TABLE statements to modify their schemas.
 type AlterableTable interface {
 	Table
 	UpdatableTable
 
-	// AddColumn adds a column to this table as given. If non-nil, order specifies where in the schema to add the column.
+	// AddColumn adds a column to this table as given. If non-nil, order
+	// specifies where in the schema to add the column.
 	AddColumn(ctx *Context, column *Column, order *ColumnOrder) error
 	// DropColumn drops the column with the name given.
 	DropColumn(ctx *Context, columnName string) error
-	// ModifyColumn modifies the column with the name given, replacing with the new column definition provided (which may
-	// include a name change). If non-nil, order specifies where in the schema to move the column.
+	// ModifyColumn modifies the column with the name given, replacing
+	// with the new column definition provided (which may include a name
+	// change). If non-nil, order specifies where in the schema to move
+	// the column.
 	ModifyColumn(ctx *Context, columnName string, column *Column, order *ColumnOrder) error
 }
 
+type SchemaValidator interface {
+	// ValidateSchema lets storage integrators validate whether they
```

Comment:
Probably needs a little more of a comment about how this is used and how implementing changes behavior of ALTER TABLE statements